ross/shaft_element.py

```diff
@@ -1192,3 +1192,1133 @@ def section(
+        theta_0 - rotation around horizontal, 
+        psi_0   - rotation around vertical, 
+        phi_0   - torsion around axial
+        x_1,u_1 - horizontal translation, 
+        y_1,v_1 - vertical translation, 
+        z_1,w_1 - axial translation, 
+        theta_1 - rotation around horizontal, 
+        psi_1   - rotation around vertical, 
+        phi_1   - torsion around axial
+        """
+        return dict(
+            u_0=0,
+            v_0=1,
+            w_0=2,
+            theta_0=3,
+            psi_0=4,
+            phi_0=5,
+            u_1=6,
+            v_1=7,
+            w_1=8,
+            theta_1=9,
+            psi_1=10,
+            phi_1=11,
+        )
+
+    def __repr__(self):
+        """This function returns a string representation of a 6 DoF shaft element.
+
+        Parameters
+        ----------
+        Returns
+        -------
+        A string representation of a 6 DoF shaft object.
+
+        Examples
+        --------
+        >>> from ross.materials import steel
+        >>> shaft1 = ShaftElement(
+        ...        L=0.25, idl=0, idr=0, odl=0.05, odr=0.08,
+        ...        material=steel, rotary_inertia=True, shear_effects=True
+        ... )
+        >>> shaft1 # doctest: +ELLIPSIS
+        ShaftElement(L=0.25, idl=0.0...
+        """
+        return (
+            f"{self.__class__.__name__}"
+            f"(L={self.L:{0}.{5}}, idl={self.idl:{0}.{5}}, "
+            f"idr={self.idr:{0}.{5}}, odl={self.odl:{0}.{5}},  "
+            f"odr={self.odr:{0}.{5}}, material={self.material.name!r}, "
+            f"alpha={self.alpha:{0}.{5}}, beta={self.beta:{0}.{5}},  "
+            f"n={self.n})"
+        )
+
+    def __str__(self):
+        """
+        Method to convert object into string
+        
+        Parameters
+        ----------
+        Returns
+        -------
+        The object's parameters translated to strings
+        """
+        return (
+            f"\nElem. N:    {self.n}"
+            f"\nLenght:     {self.L:{10}.{5}}"
+            f"\nLeft Int. Diam.: {self.idl:{10}.{5}}"
+            f"\nLeft Out. Diam.: {self.odl:{10}.{5}}"
+            f"\nRight Int. Diam.: {self.idr:{10}.{5}}"
+            f"\nRight Out. Diam.: {self.odr:{10}.{5}}"
+            f"\nAlpha damp. factor: {self.alpha:{10}.{5}}"
+            f"\nBeta damp. factor: {self.beta:{10}.{5}}"
+            f'\n{35*"-"}'
+            f"\n{self.material}"
+            f"\n"
+        )
+
+    def M(self):
+        r"""Mass matrix for an instance of a 6 DoF shaft element.
+
+        Returns
+        -------
+        M: np.ndarray
+            Mass matrix for the 6 DoF shaft element.
+
+        Examples
+        --------
+        >>> Timoshenko_Element = ShaftElement(0.25, 0, 0.05, steel,
+        ...                                  rotary_inertia=True,
+        ...                                  shear_effects=True)
+        >>> Timoshenko_Element.M()[:12, :12]
+        array(12x12)
```

Comment:
I think the output can not be written this way.
An alternative is:
```
        >>> Timoshenko_Element = ShaftElement(0.25, 0, 0.05, steel,
        ...                                  rotary_inertia=True,
        ...                                  shear_effects=True)
        >>> Timoshenko_Element.M()[:12, :12].shape
        (12, 12)
```
or you can do like this one below (do not copy the output because its from another example):
```
        >>> Timoshenko_Element = ShaftElement(
        ...                         L=0.5, idl=0.05, idr=0.05, odl=0.1,
        ...                         odr=0.15, material=steel,
        ...                         rotary_inertia=True,
        ...                         shear_effects=True)
        >>> Timoshenko_Element.M()[:4, :4]
        array([[11.36986417,  0.        ,  0.        ,  0.86197637],
               [ 0.        , 11.36986417, -0.86197637,  0.        ],
               [ 0.        , -0.86197637,  0.08667495,  0.        ],
               [ 0.86197637,  0.        ,  0.        ,  0.08667495]])
```